pkg/fileutil/fileutil_linux.go

```diff
@@ -105,3 +106,35 @@ func Lsetxattr(path string, attr string, data []byte, flags int) error {
 	}
 	return nil
 }
+
+// GetDeviceInfo returns the type, major, and minor numbers of a device
+func GetDeviceInfo(path string) (kind rune, major uint64, minor uint64, err error) {
```

Comment:
Change this to

```
func GetDeviceInfo(path string) (kind rune, major uint64, minor uint64, err error) {
   	d, err := os.Lstat(path)
  	if err != nil {
  		return ' ', 0, 0, err
  	}
        return getDeviceInfo(d)
}

func getDeviceInfo(d *os.FileInfo) (kind rune, major uint64, minor uint64, err error) {
...
}
```

Then you can test the inner function to your heart's content